File: B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/IA/take_object.py
from mouv import handle_movement

def take_action(s, ia, tab):
    if ia.meetup == True and "player" in tab:
        return
    for element in tab:
        if "food" == element:
            take_food(s, ia)
        if ("linemate" == element and ia.state == "FOOD_MOD") or ("linemate" == element and ia.state == "LINEMATE_MOD" and ia.nb_global_linemate < 8):
            take_linemate(s, ia)
        if ("deraumere" == element and ia.state == "FOOD_MOD") or ("deraumere" == element and ia.state == "DERAUMERE_MOD" and ia.nb_global_deraumere < 8):
            take_deraumere(s, ia)
        if ("sibur" == element and ia.state == "FOOD_MOD") or ("sibur" == element and ia.state == "SIBUR_MOD" and ia.nb_global_sibur < 10):
            take_sibur(s, ia)
        if ("mendiane" == element and ia.state == "FOOD_MOD") or ("mendiane" == element and ia.state == "MENDIANE_MOD" and ia.nb_global_mendiane < 5):
            take_mendiane(s, ia)
        if ("phiras" == element and ia.state == "FOOD_MOD") or ("phiras" == element and ia.state == "PHIRAS_MOD" and ia.nb_global_phiras < 6):
            take_phiras(s, ia)
        if ("thystame" == element and ia.state == "FOOD_MOD") or ("thystame" == element and ia.state == "THYSTAME_MOD" and ia.nb_global_thystame < 1):
            take_thystame(s, ia)
        handle_movement(s, ia)

def take_food(s, ia):
    ia.steps.append('Take food')
    print("reponse take food:")

def take_linemate(s, ia):
    ia.steps.append('Take linemate')
    print("reponse take linemate:")
    ia.nb_global_linemate += 1

def take_deraumere(s, ia):
    ia.steps.append('Take deraumere')
    print("reponse take deraumere:")
    ia.nb_global_deraumere += 1

def take_sibur(s, ia):
    ia.steps.append('Take sibur')
    print("reponse take sibur:")
    ia.nb_global_sibur += 1

def take_mendiane(s, ia):
    ia.steps.append('Take mendiane')
    print("reponse take mendiane:")
    ia.nb_global_mendiane += 1

def take_phiras(s, ia):
    ia.steps.append('Take phiras')
    print("reponse take phiras:")
    ia.nb_global_phiras += 1

def take_thystame(s, ia):
    ia.steps.append('Take thystame')
    print("reponse take thystame:")
    ia.nb_global_thystame += 1

File: B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/ai/src/take_object.cpp
#include "take_object.hpp"
#include "mouv.hpp"
#include "IA.hpp"
#include <iostream>
#include <algorithm>
#include "MyIA.hpp"
void take_action(int s, MyIA& ia, const std::vector<std::string>& tab) {
    if (ia.meetup == true && std::find(tab.begin(), tab.end(), "player") != tab.end()) {
        return;
    }
    for (const auto& element : tab) {
        if (element == "food") {
            take_food(ia);
        }
        if ((element == "linemate" && ia.state == "FOOD_MOD") || 
            (element == "linemate" && ia.state == "LINEMATE_MOD" && ia.nb_global_linemate < 8)) {
            take_linemate(ia);
        }
        if ((element == "deraumere" && ia.state == "FOOD_MOD") || 
            (element == "deraumere" && ia.state == "DERAUMERE_MOD" && ia.nb_global_deraumere < 8)) {
            take_deraumere(ia);
        }
        if ((element == "sibur" && ia.state == "FOOD_MOD") || 
            (element == "sibur" && ia.state == "SIBUR_MOD" && ia.nb_global_sibur < 10)) {
            take_sibur(ia);
        }
        if ((element == "mendiane" && ia.state == "FOOD_MOD") || 
            (element == "mendiane" && ia.state == "MENDIANE_MOD" && ia.nb_global_mendiane < 5)) {
            take_mendiane(ia);
        }
        if ((element == "phiras" && ia.state == "FOOD_MOD") || 
            (element == "phiras" && ia.state == "PHIRAS_MOD" && ia.nb_global_phiras < 6)) {
            take_phiras(ia);
        }
        if ((element == "thystame" && ia.state == "FOOD_MOD") || 
            (element == "thystame" && ia.state == "THYSTAME_MOD" && ia.nb_global_thystame < 1)) {
            take_thystame(ia);
        }
        handle_movement(s, ia);
    }
}

void take_food(MyIA& ia) {
    ia.steps.push_back("Take food");
    std::cout << "reponse take food:" << std::endl;
}

void take_linemate(MyIA& ia) {
    ia.steps.push_back("Take linemate");
    std::cout << "reponse take linemate:" << std::endl;
    ia.nb_global_linemate++;
}

void take_deraumere(MyIA& ia) {
    ia.steps.push_back("Take deraumere");
    std::cout << "reponse take deraumere:" << std::endl;
    ia.nb_global_deraumere++;
}

void take_sibur(MyIA& ia) {
    ia.steps.push_back("Take sibur");
    std::cout << "reponse take sibur:" << std::endl;
    ia.nb_global_sibur++;
}

void take_mendiane(MyIA& ia) {
    ia.steps.push_back("Take mendiane");
    std::cout << "reponse take mendiane:" << std::endl;
    ia.nb_global_mendiane++;
}

void take_phiras(MyIA& ia) {
    ia.steps.push_back("Take phiras");
    std::cout << "reponse take phiras:" << std::endl;
    ia.nb_global_phiras++;
}

void take_thystame(MyIA& ia) {
    ia.steps.push_back("Take thystame");
    std::cout << "reponse take thystame:" << std::endl;
    ia.nb_global_thystame++;
}